obelisk/cpp/obelisk_cpp/include/obelisk_sim_robot.h

```diff
@@ -0,0 +1,139 @@
+#pragma once
+
+#include "obelisk_robot.h"
+#include "obelisk_sensor_msgs/msg/true_sim_state.hpp"
+
+namespace obelisk {
+    template <typename ControlMessageT> class ObeliskSimRobot : public ObeliskRobot<ControlMessageT> {
+        using TrueSimState = obelisk_sensor_msgs::msg::TrueSimState;
+
+      public:
+        explicit ObeliskSimRobot(const std::string& name, const std::string& state_timer_key = "state_timer",
+                                 const std::string& state_pub_key = "state_pub")
+            : ObeliskRobot<ControlMessageT>(name), state_timer_key_(state_timer_key), state_pub_key_(state_pub_key) {
+
+            // Register Components
+            this->RegisterTimer("timer_true_sim_state_setting", state_timer_key_,
+                                std::bind(&ObeliskSimRobot::PublishTrueSimState, this));
+            this->template RegisterPublisher<TrueSimState>("pub_true_sim_state_setting", state_pub_key_);
+
+            stop_thread_ = false;
+        }
+
+        /**
+         * @brief Configures all the required ROS components. Specifcially this
```

Comment:
sp: specifically

we should somehow add a spellchecker

Reply:
I'll look into it... spelling isn't my strong suit.

Reply:
Leaving this here for my own reference: https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker